python/composio/tools/base/runtime.py

```diff
@@ -253,12 +253,23 @@ def _parse_docstring(
     return header, params, returns
 
 
+def _get_connected_account(app: str, entity_id: str) -> t.Optional[ConnectedAccountModel]:
+    try:
+        client = Composio.get_latest()
+        connected_account = client.connected_accounts.get(
+            connection_id=client.get_entity(entity_id).get_connection(app=app).id
+        )
+        return connected_account
+    except ComposioClientError:
+        return None
+
 def _get_auth_params(app: str, entity_id: str) -> t.Optional[t.Dict]:
```

Comment:
Consider extracting the common code between `_get_connected_account` and `_get_auth_params` into a shared helper function since they share similar initialization and error handling patterns. This would reduce code duplication and make future maintenance easier.